src/common.hpp

```diff
@@ -554,4 +554,35 @@ class Stopwatch
     std::chrono::high_resolution_clock::time_point tstart;
 };
 
+////////////////////////////////////////////////////////////////////////////////
+// A global log that can stream to an external source (like unreal).
+// Intended to replace calls to printf when stdout is not available.
+// Not thread safe or intended for speed.
+////////////////////////////////////////////////////////////////////////////////
+
+static void (*EXTERNAL_GLOBAL_LOG)(const char* message) = nullptr;
```

Comment:
Not trying to beat a dead horse here, but: Since this is declared static in a common header, a separate copy of it will be created in each translation unit. But only the one in `setup.cpp` will get set to the Unreal callback; all the others will be `nullptr` and so most of the log messages will still go to `stdout`. Fixing this in my version too.